### DISCUSSION TITLE
Play and Scala 3

### DISCUSSION THREAD

**Discussion by gavinbaumanis** [playframework/playframework#11242] [General]
Hi There, 
Just wondering if there was an appropriate place I should be visiting / subscribed to - to get updates, specifically, about Play for Scala 3, please.

Thanks.

**Comment by raul782** [comment]
The last update that I found is here, https://discuss.lightbend.com/t/scala-3-0-support/7729/10
Maybe we can convert that post into an actionable issue?

**Reply by ihostage** [reply]
You are right. As Matthias @mkurz said
> There is work done towards adding Scala 3 support for Play, distributed in different places.
It’s not as easy as simply making Play itself work with Scala 3. Play depends on various other projects and libraries. 

As I can see the current state is:
* ✅ **Akka 2.6.18** already support Scala 3 (https://akka.io/blog/news/2021/12/20/akka-2.6.18-released)
* :stop_sign: **Akka HTTP** don't support Scala3  (https://github.com/akka/akka-http/issues/3891)
* 🕐 **Play-WS 2.2.0** will be support Scala 3 (https://github.com/playframework/play-ws/pull/604)
* 🕐 **Play-JSON 2.10.0** will be support Scala 3 (but since [**2.10.0-RC5**](https://github.com/playframework/play-json/releases/tag/2.10.0-RC5) already support)
* 🕐 **Twirl 1.6.0** will be support Scala 3 (https://github.com/playframework/twirl/pull/461)

Without these libraries we can't support Scala 3 for core of Play 🤷‍♂️

**Reply by mkurz** [reply]
akka-http is getting closer to Scala 3 every day:
* https://github.com/akka/akka-http/pull/4097
* https://github.com/akka/akka-http/pull/4100
* https://github.com/akka/akka-http/pull/4101
* https://github.com/akka/akka-http/pull/4108
* https://github.com/akka/akka-http/pull/4107
* https://github.com/akka/akka-http/pull/4109
* https://github.com/akka/akka-http/pull/4079

Also, looks like a local snapshot of akka-http Scala 3 is used already here as well: https://github.com/akka/akka-grpc/pull/1608
So I guess it will be ready soon.

**Reply by ghost** [reply]
What is the latest status regarding the support of Scala 3 from Play ?

**Reply by mkurz** [reply]
> What is the latest status regarding the support of Scala 3 from Play ?

Play 2.9 and 3.0 with Scala 3 support have been released meanwhile: https://twitter.com/playframework/status/1721576472047911007
